modules/tidb/src/main/java/org/testcontainers/containers/TiDBContainer.java

```diff
@@ -0,0 +1,129 @@
+package org.testcontainers.containers;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Icemap
+ */
+public class TiDBContainer extends JdbcDatabaseContainer<TiDBContainer> {
+
+    public static final String NAME = "tidb";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("pingcap/tidb");
+
+    private static final String IMAGE = DEFAULT_IMAGE_NAME.getUnversionedPart();
+
+    public static final Integer TIDB_PORT = 4000;
+
+    private static final int REST_API_PORT = 10080;
+
+    private String databaseName = "test";
+
+    private String username = "root";
+
+    private String password = "";
+
+    public TiDBContainer(String tag) {
+        this(DockerImageName.parse(TiDBContainer.IMAGE).withTag(tag));
+    }
+
+    public TiDBContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        addExposedPorts(TIDB_PORT, REST_API_PORT);
```

Comment:
both ports should be exposed? on only `4000`?

Reply:
Both ports should be exposed. `TIDB_PORT` is 4000, and `REST_API_PORT` is 10080.